src/App/Fossa/LicenseScanner.hs

```diff
@@ -0,0 +1,119 @@
+import App.Fossa.FossaAPIV1 qualified as Fossa
+import Control.Effect.Diagnostics (Diagnostics, context, fatalText)
+import Control.Monad (unless)
+import Crypto.Hash (Digest, MD5, hash)
+import Data.ByteString qualified as B
+import Data.List (nub)
+import Effect.Exec (Exec, runExecIO)
+import Fossa.API.Types
+import Path
+import Srclib.Types (
+  LicenseSourceUnit (..),
+  LicenseUnit (..),
+  Locator (..),
+ )
+
+import Data.String.Conversion (encodeUtf8, toString, toText)
+import Data.Text (Text)
+
+import App.Fossa.EmbeddedBinary (ThemisBins, withThemisAndIndex)
+
+runLicenseScanOnDir ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  ) =>
+  Path Abs Dir ->
+  m [LicenseUnit]
+runLicenseScanOnDir scanDir = withThemisAndIndex $ themisRunner scanDir
+
+themisRunner ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  ) =>
+  Path Abs Dir ->
+  ThemisBins ->
+  m [LicenseUnit]
+themisRunner scanDir themisBins = do
+  runExecIO $ runThemis themisBins scanDir
+
+runThemis :: (Has Exec sig m, Has Diagnostics sig m) => ThemisBins -> Path Abs Dir -> m [LicenseUnit]
+runThemis themisBins scanDir = do
+  context "Running license scan binary" $ execThemis themisBins scanDir
+
+md5 :: B.ByteString -> Digest MD5
+md5 = hash
+
+scanAndUploadVendoredDep :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m, Has StickyLogger sig m) => ApiOpts -> Path Abs Dir -> VendoredDependency -> m Archive
+scanAndUploadVendoredDep apiOpts baseDir VendoredDependency{..} = context "compressing and uploading vendored deps" $ do
+  logSticky $ "License Scanning '" <> vendoredName <> "' at '" <> vendoredPath <> "'"
+  vendoredDepDir <- case parseRelDir $ toString vendoredPath of
+    Left _ -> fatalText "Error constructing scan dir for vendored scan"
+    Right val -> pure val
+  let cliScanDir = baseDir </> vendoredDepDir
+  themisScanResult <- runLicenseScanOnDir cliScanDir
+  let licenseSourceUnit =
+        LicenseSourceUnit
+          { licenseSourceUnitName = vendoredPath
+          , licenseSourceUnitType = "cli-license-scanned"
+          , licenseSourceUnitLicenseUnits = themisScanResult
+          }
+
+  depVersion <- case vendoredVersion of
+    Nothing -> pure $ toText $ show $ md5 $ encodeUtf8 $ show themisScanResult
+    Just version -> pure version
+
+  signedURL <- Fossa.getSignedLicenseScanURL apiOpts depVersion vendoredName
+
+  logSticky $ "Uploading '" <> vendoredName <> "' to secure S3 bucket"
+  _ <- Fossa.licenseScanResultUpload signedURL licenseSourceUnit
+
+  pure $ Archive vendoredName depVersion
+
+-- archiveUploadSourceUnit receives a list of vendored dependencies, a root path, and API settings.
+-- Using this information, it license scans each vendored dependency, uploads the license scan results and then queues a build for the dependency.
+licenseScanSourceUnit :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m, Has StickyLogger sig m) => Path Abs Dir -> ApiOpts -> [VendoredDependency] -> m [Locator]
```

Comment:
I think we should do this, but in a separate PR where we avoid hitting either the archive upload or the cli license scan endpoints if `vendoredDeps` is empty.

I think this would involve changing the definition of `ManualDependencies` to something like this:

```haskell
data ManualDependencies = ManualDependencies
  { referencedDependencies :: Maybe (NotEmpty ReferencedDependency)
  , customDependencies :: Maybe (NotEmpty CustomDependency)
  , vendoredDependencies :: Maybe (NotEmpty VendoredDependency)
  , remoteDependencies :: Maybe (NotEmpty RemoteDependency)
  }
```